ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/actions/CreateProjectAction.java

```diff
@@ -34,26 +38,42 @@
 
     private final ProjectWizardPresenter wizard;
     private final AnalyticsEventLogger   eventLogger;
-    private final AppContext             appContext;
+    private final DtoFactory dtoFactory;
 
     @Inject
-    public CreateProjectAction(Resources resources, ProjectWizardPresenter wizard, AnalyticsEventLogger eventLogger, AppContext appContext) {
+    public CreateProjectAction(Resources resources,
+                               ProjectWizardPresenter wizard,
+                               AnalyticsEventLogger eventLogger,
+                               DtoFactory dtoFactory) {
         super(Arrays.asList(PROJECT_PERSPECTIVE_ID), "Create Project...", "Create new project", null, resources.newProject());
         this.wizard = wizard;
         this.eventLogger = eventLogger;
-        this.appContext = appContext;
+        this.dtoFactory = dtoFactory;
     }
 
     @Override
     public void actionPerformed(ActionEvent e) {
         eventLogger.log(this);
-        wizard.show();
+        final Map<String, String> parameters = e.getParameters();
+        if (parameters != null && !parameters.isEmpty()) {
+            final ProjectConfigDto dataObject = dtoFactory.createDto(ProjectConfigDto.class);
+            if (parameters.containsKey("projectName")) {
+                dataObject.setName(parameters.get("projectName"));
+            }
+
+            if (parameters.containsKey("projectType")) {
+                dataObject.setType(parameters.get("projectType"));
+            } else {
+                dataObject.setType("blank");
```

Comment:
org.eclipse.che.api.project.shared.Constants#BLANK_ID instead of hardcoded string